docs/src/rfc/1-pivotkey.md

```diff
@@ -0,0 +1,130 @@
+This is helpful in a number of scenarios such as data prefetching or
+disk-to-disk migrations.
+
+Since pivots are stored in the internal nodes we are required to read a
+substantial amount of data to retrieve knowledge about all exisiting pivot keys.
+This limits the efficient usage to scenarios in which we retrieve pivot keys,
+for example from messages emitted by the DMU, and record these as they are used
+by the user. Previously a similar scheme has been done by the migration policies
+which recorded disk offsets and set hints to the DMU to which tier a node is
+advised to be written.
+This limited hints to often accessed nodes which are likely to be migrated to
+faster storage, as not often accessed nodes would not encounter the sent hints.
+With Pivot Keys we could actively migrate them downwards, which is the main
+advantage of Pivot Keys. This can be useful in scenarios with additional small
+layers like NVRAM where we are then more flexible to migrate granular amounts of
+data for better tier utilization.
+
+# Drawbacks
+
+To be fully usable in multiple layers the Pivot Key is required to be stored in
+each node taking up (at the moment) an arbitrary amout of extra space depending
+on the pivot element. The DML can then extract Pivot Keys from each node it acts
+on and report critical actions such as fetch, write and delete. This results in
+more code in alot of places of the `tree` and `data_management` modules as these
+will have to be adjusted to accomodate for the extra encoded member.
+
+# Alternatives
+
+The search for alternatives is difficult as not many characteristics are present
+which allow nodes to be identifiable *and* searchable. Something like Ids for
+example could make nodes identifiable but from an Id we cannot search for the
+node in the tree.
+
+An alternative to the pivot method could be the path construction which relies
+on the structural uniqueness of the tree, in which we construct a path from the
+root by giving directions for each step downwards until the desired node is
+located. It can look similar to this:
+
+```rust
+enum Path {
+    Child(usize, Path),
+    Terminate,
+}
+```
+
+This method does not provide many advantages and carries the disadvantage of
+having to maintain additional state when constructing keys about the just
+traversed distance. Arguably, this is semantic-wise not complicated but many
+methods will be affected by this change. Also, misidentification may become
+possible as with reconstruction of subtrees paths may be shifted around. With
+Pivot Keys these will result in a failed search indicating an outdated key.
+Currently the only advantage this method would have to the Pivot Key method is
+that the size can be expected to remain comparatively low, with 5 bytes for each
+element in the search path. A restriction of key size as discussed in
+[Issue](https://github.com/julea-io/haura/issues/12) could solve this problem
```

Comment:
"issue 12" or "the corresponding issue"?

Reply:
Corresponding Issue sounds good